pkgs/by-name/gl/globalprotect-openconnect_2/package.nix

```
@@ -0,0 +1,138 @@
+{
+  perl,
+  jq,
+  fetchFromGitHub,
+  lib,
+  openconnect,
+  libsoup,
+  webkitgtk_4_0,
+  pkg-config,
+  callPackage,
+  rustPlatform,
+  glib,
+  atk,
+  gdk-pixbuf,
+  pango,
+  cairo,
+  harfbuzz,
+  gtk3,
+  zlib,
+  makeDesktopItem,
+  vpnc-scripts,
+  includeUnfreeGui ? false,
+}:
+let
+  version = "2.3.9";
+  pname = "globalprotect-openconnect";
+  desktop = makeDesktopItem {
+    name = "GlobalProtect Openconnect VPN Client";
+    desktopName = "GlobalProtect Openconnect VPN Client";
+    comment = "A GUI for GlobalProtect VPN";
+    genericName = "GlobalProtect VPN Client";
+    exec = "gpclient launch-gui %u";
+    icon = "gpgui";
+    categories = [
+      "Network"
+      "Dialup"
+    ];
+    mimeTypes = [ "x-scheme-handler/globalprotectcallback" ];
+  };
+
+  gpgui = if includeUnfreeGui then (callPackage ./gui.nix { inherit version; }) else null;
```

```suggestion
  gpgui = callPackage ./gui.nix { inherit version; };
```
putting null on variables is bad practice

---

> if withGui is set to false, it is not used at all.

Lazy evaluation is nice 🙂. I missed that point in my review. Thanks for catching that.